### ISSUE TITLE
User management depends on being a CouchDB admin

### ISSUE THREAD

**Issue by tlvince** [eHealthAfrica/direct-delivery-dashboard#169]

**Comment by tlvince** [comment]
Middleware approach https://github.com/tlvince/recliner

**Comment by tlvince** [comment]
Kazana/Hoodie/React approach: https://github.com/eHealthAfrica/kazana-account-app

**Comment by tlvince** [comment]
@gr2m's (hey Gregor! :wave:) thing https://github.com/gr2m/couchdb-user-management-app

**Comment by tlvince** [comment]
Bearer plugin for Hapi (use Hapi as a reverse proxy): https://github.com/eHealthAfrica/hapi-couchdb-auth-bearer-plugin

**Comment by tlvince** [comment]
More Hoodie bits:
- https://github.com/hoodiehq/hapi-couchdb-account
- https://github.com/hoodiehq/account-client

**Comment by tlvince** [comment]
Corresponding issue for Call Centre (and GN connect): https://github.com/eHealthAfrica/sl-ebola-call-admin/issues/1026

**Comment by tlvince** [comment]
@jofomah do you think we can get away with having user management in a separate app? Maybe `admin-dd.ehealth.org.ng`?

**Comment by tlvince** [comment]
@janetbutts think we'll need to look at this before releasing. In short, if CouchDB's security is set up properly, then users page will not work, including for `super` users unless the user is a "real" CouchDB admin user. Handing out CouchDB admin credentials is a bad idea security wise. One immediate solution is to introduce a middleware layer between the app and CouchDB that proxies user management actions. ~~Otherwise, we have a [generic CouchDB user management app](http://gr2m.github.io/couchdb-user-management-app/) that could be repurposed, but would need to live on its own address e.g. `admin-dd.ehealth.org.ng`.~~ (edit: this will still require giving CouchDB admin credentials to our super users).

**Comment by janetbutts** [comment]
@tlvince , agreed that we do not want dashboard users to also be couchdb admins. what is the effort involved with this middleware layer? is this a show stopper for the release?

**Comment by tlvince** [comment]
Most of the middleware service work preexists (see above), so I don't estimate more than a day of integration. However, coordination with ops for deployment may take some time, depending on their schedule.

That said, the fact that this currently works in dev suggests to me CouchDB's security has not been set up correctly. I will review and also check stage and prod.

**Comment by tlvince** [comment]
Working on this.

**Comment by janetbutts** [comment]
@tlvince, this will get tabled until the next release. for right now, we will just hide the security tab (#321).

**Comment by tlvince** [comment]
Un-assigning for now.
